plugins/discovery-gce/src/main/java/org/elasticsearch/cloud/gce/network/GceNameResolver.java

```diff
@@ -109,8 +111,9 @@ public GceNameResolver(Settings settings, GceComputeService gceComputeService) {
             if (metadataResult == null || metadataResult.length() == 0) {
                 throw new IOException("no gce metadata returned from [" + gceMetadataPath + "] for [" + value + "]");
             }
-            // only one address: because we explicitly ask for only one via the GceHostnameType
-            return new InetAddress[] { InetAddress.getByName(metadataResult) };
+            // really should be only one address: because we explicitly ask for only one via the Ec2HostnameType
+            // but why do we even allow configuring this by hostname...
```

Comment:
I totally agree with you. Actually when I wrote the code, I did the same thing which was done for EC2 plugin.
I think that we can totally remove that and only support IP address. 

It can be done in a follow up issue though.